pkg/asset/cluster/azure/azure.go

```diff
@@ -30,13 +31,58 @@ func Metadata(config *types.InstallConfig) *azure.Metadata {
 // PreTerraform performs any infrastructure initialization which must
 // happen before Terraform creates the remaining infrastructure.
 func PreTerraform(ctx context.Context, clusterID string, installConfig *installconfig.InstallConfig) error {
-	if len(installConfig.Config.Azure.ResourceGroupName) == 0 {
+	session, err := installConfig.Azure.Session()
+	if err != nil {
+		return fmt.Errorf("failed to get session: %w", err)
+	}
+
+	if err := tagResourceGroup(ctx, clusterID, installConfig, session); err != nil {
+		return err
+	}
+
+	err = tagVNet(ctx, clusterID, installConfig, session)
+	return err
+}
+
+func tagVNet(ctx context.Context, clusterID string, installConfig *installconfig.InstallConfig, session *icazure.Session) error {
+	if len(installConfig.Config.Azure.VirtualNetwork) == 0 {
 		return nil
 	}
 
-	session, err := installConfig.Azure.Session()
+	resourceGroupName := installConfig.Config.Azure.NetworkResourceGroupName
+
+	vnetClient, err := armnetwork.NewVirtualNetworksClient(session.Credentials.SubscriptionID, session.TokenCreds, nil)
+	if err != nil {
+		return fmt.Errorf("failed to get the virtual network client: %w", err)
+	}
+
+	vnetResp, err := vnetClient.Get(ctx, resourceGroupName, installConfig.Config.Azure.VirtualNetwork, nil)
 	if err != nil {
-		return errors.Wrap(err, "failed to get session")
+		return fmt.Errorf("failed to get the virtual network %q: %w", installConfig.Config.Azure.VirtualNetwork, err)
+	}
+	vnet := vnetResp.VirtualNetwork
+	if vnet.Tags == nil {
+		vnet.Tags = map[string]*string{}
+	}
+	tagKey, tagValue := sharedTag(clusterID)
+	vnet.Tags[tagKey] = tagValue
+
+	tags := armnetwork.TagsObject{
+		Tags: vnet.Tags,
+	}
+
+	if _, err := vnetClient.UpdateTags(
```

Comment:
```suggestion
    logrus.Debugf("Tagging %s with kubernetes.io/cluster/%s: shared", installConfig.Config.Azure.VirtualNetwork, clusterID)
	if _, err := vnetClient.UpdateTags(
```
Let's add a log here like it's done for the resource group.

Reply:
done